stack/__main__.py

```diff
@@ -331,13 +331,76 @@ def create_group(mail: str) -> gcp.cloudidentity.Group:
         member=f'serviceAccount:{WEB_SERVER_SERVICE_ACCOUNT}',
     )
 
-    for group_prefix in ACCESS_LEVELS:
-        gcp.secretmanager.SecretIamMember(
-            f'sample-metadata-api-{group_prefix}-access-group-cache-secret-accessor',
-            secret_id=access_group_cache_secrets[group_prefix].id,
-            role='roles/secretmanager.secretAccessor',
-            member=f'serviceAccount:{SAMPLE_METADATA_API_SERVICE_ACCOUNT}',
-        )
+    # Sample metadata access
+    #   - 4 secrets, production-read, production-write, development-read, development-write
+    sm_groups = {}
+    for env in ('production', 'test'):
+        for rs in ('read', 'write'):
+            key = f'sample-metadata-{env}-{rs}'
+            secret_id = f'{dataset}-{key}-members-cache'
+
+            group = create_group(group_mail(dataset, key))
+            sm_groups[f'{env}-{rs}'] = group
+
+            gcp.cloudidentity.GroupMembership(
+                f'sample-metadata-group-cache-{env}-{rs}-access-level-group-membership',
```

Comment:
This is not directly tied to an access level anymore, maybe we can remove that from the name?

Reply:
Yep, makes a lot of sense. I've fixed a lot the names across sample-metadata permissions here.